packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -2209,6 +2209,60 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('foo'), findsOneWidget);
   });
+
+  testWidgets('The opened menu should follow if the button\'s position changed', (WidgetTester tester) async {
+    final GlobalKey buttonKey = GlobalKey();
+
+    Widget buildFrame(double width, double height) {
+      return MaterialApp(
+        home: Scaffold(
+          body:  SizedBox(
+            height: height,
+            width: width,
+            child: Center(
+              child: PopupMenuButton<int>(
+                child: SizedBox(
+                  key: buttonKey,
+                  height: 10.0,
+                  width: 10.0,
+                  child: const ColoredBox(
+                    color: Colors.pink,
+                  ),
+                ),
+                itemBuilder: (BuildContext context) => <PopupMenuEntry<int>>[
+                  const PopupMenuItem<int>(child: Text('-1-'), value: 1,),
+                  const PopupMenuItem<int>(child: Text('-2-'), value: 2,),
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(100.0, 100.0));
+
+    // Open the menu.
+    await tester.tap(find.byKey(buttonKey));
+    await tester.pumpAndSettle();
+
+    Offset button = tester.getTopLeft(find.byKey(buttonKey));
```

Comment:
Here and elsewhere - Clearer variable name
```suggestion
    Offset buttonOffset = tester.getTopLeft(find.byKey(buttonKey));
```